.github/workflows/ci.yaml

```diff
@@ -112,21 +112,74 @@ jobs:
     name: Check operator CRDs
     runs-on: ubuntu-latest
     steps:
-    - name: Checkout
-      uses: actions/checkout@v4
-    - name: Make manifests
-      env:
-        GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      run: make -C operator manifests
-    - name: Check CRDs
-      run: |
-        git diff --exit-code --name-only
-        if [ $? -eq 0 ]; then
-          echo "CRDs are up to date"
-        else
-          echo "CRDs are out of date"
-          exit 1
-        fi
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
+          cache-dependency-path: "**/*.sum"
+      - name: Make manifests
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: make -C operator manifests
+      - name: Check CRDs
+        run: |
+          git diff --exit-code --name-only
+          if [ $? -eq 0 ]; then
+            echo "CRDs are up to date"
+          else
+            echo "CRDs are out of date"
+            exit 1
+          fi
+
+  check-swagger-docs:
+    name: Check swagger docs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
+          cache-dependency-path: "**/*.sum"
+      - name: Check swagger docs
+        run: |
+          make -C api swagger
+          git diff --exit-code --name-only
+          if [ $? -eq 0 ]; then
+            echo "Swagger docs are up to date"
+          else
+            echo "Swagger docs are out of date"
+            exit 1
+          fi
+
+  check-web:
+    name: Check web
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup node
+        uses: actions/setup-node@v4
+        with:
+          node-version: lts/*
+      - name: Install dependencies
+        working-directory: ./web
+        run: npm i
+      - name: Build
+        working-directory: ./web
+        run: npm run build
+      - name: Check web is up to date
+        run: |
+          git diff --exit-code --name-only
+          if [ $? -eq 0 ]; then
+            echo "Web is up to date"
+          else
+            echo "Web is out of date"
+            exit 1
+          fi
```

Comment:
Alternatively, why don't we build the web assets on CI VS having to locally build it and commit it? Things such as `npm install` are non-deterministic, meaning my local web build can be different from yours. In CI we can make sure we have reproducible builds via specific node+npm versions and commands such as `npm ci` for example.